File: easy/724. Find Pivot Index/724. Find Pivot Index.go
package main

func countSum(nums []int) int {
	sum := 0
	for _, num := range nums {
		sum += num
	}
	return sum
}

func pivotIndex(nums []int) int {
	total_sum := countSum(nums)
	n := len(nums)
	leftSum := 0

	for i := 0; i < n; i++ {
		rightSum := total_sum - nums[i] - leftSum
		if rightSum == leftSum {
			return i
		}
		leftSum += nums[i]
	}

	return -1
}

File: easy/724. Find Pivot Index/724. Find Pivot Index.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total_sum = sum(nums)

        n = len(nums)
        left_sum = 0

        for i in range(n):
            right_sum = total_sum - nums[i] - left_sum
            if left_sum == right_sum:
                return i
            left_sum += nums[i]

        return -1
